src/AspNetCoreModuleV2/CommonLib/application.h

```diff
@@ -20,24 +19,38 @@ class APPLICATION : public IAPPLICATION
     APPLICATION_STATUS
     QueryStatus() override
     {
-        return m_status;
+        return m_fStopCalled ? APPLICATION_STATUS::RECYCLED : APPLICATION_STATUS::RUNNING;
     }
 
     APPLICATION()
-        : m_cRefs(1)
+        : m_fStopCalled(false),
+          m_cRefs(1)
     {
         InitializeSRWLock(&m_stateLock);
     }
 
-    
+
     VOID
     Stop(bool fServerInitiated) override
     {
-        UNREFERENCED_PARAMETER(fServerInitiated);
+        SRWExclusiveLock stopLock(m_stateLock);
+
+        if (m_fStopCalled)
+        {
+            return;
+        }
 
         m_fStopCalled = true;
+
+        StopInternal(fServerInitiated);
+    }
+
+    virtual
+    VOID
+    StopInternal(bool fServerInitiated)
```

Comment:
Feels weird to have an `Internal` function as virtual.

Reply:
DoStop? StopImpl?

Reply:
Make this one `Stop` and make the call in Application.cpp `StopApplication`?

Reply:
Is it better?

Reply:
Not enough to actually make that change.